### DISCUSSION TITLE
Please add support for items dated to portions of decade on Timeline

### DISCUSSION THREAD

**Discussion by tprus-luc** [CollectionBuilder/collectionbuilder.github.io#137] [Ideas]
Many of the items in my project can only be dated to an approximate half decade, or even only to the approximate decade. CB seems confused by this.

To some extent, CB's timeline handles these. For example, if I enter "1920s" in the date column of my CSV data file, CB puts "1920s" items before items dated to 1920. This is acceptable.

But if I enter "early 1920s" or "late 1920s" CB does not put those item before and after "1920s" respectively, as one would expect. Instead it places both sets of items at the bottom of the Timeline page, after all other items. This is nonintuitive

Entering "1920s early" or "1920s late" is a little better, as CB puts these items after 1920 and 1920s.

Putting it all together, CB sorts items like this: 1920, 1920s, 1920s early, 1920s late, 1921, 1922 .... 2024, early 1920s, late 1920s

Please add support for these approximate dates, or, please let me know the best way to enter dates for items that can't be dated to a year.

**Comment by evanwill** [comment]
All the logic for things like the Timeline layout is using Liquid templating language in Jekyll, which is extremely limited--so complicated parsing of free text dates isn't really possible. The idea is to do that sort of parsing via your metadata work, using tools at your disposal such as OpenRefine or equations in spreadsheets to generate a consistent date field that you can use for your desired purposes. Creating a Timeline sorted by era or decade is a pretty common customization, but relies on metadata work (rather than parsing in the template code). Tweaking and repurposing the timeline to sort this new metadata is covered a bit in the docs, https://collectionbuilder.github.io/cb-docs/docs/advanced/repurposetimeline/

In our practice, we use an "archival_date" field where we express the full date following archival description standards including uncertainty. Then we convert that to the "date" field which is strictly ISO (yyyy, yyyy-mm, or yyyy-mm-dd). Depending on the collection context, that means making judgements about uncertainty like "early 1920s" = "1920", or if the date is uncertain, leaving it blank. On the individual Item page we display the "archival_date", but the "date" field is used for Timeline and sorting etc which helps users find relevant materials even if it isn't exact. In collections where exact dates are more critical, we leave the "date" blank to avoid putting uncertain items into the timelines.
